lib/features/email/data/network/email_api.dart

```diff
@@ -245,36 +244,52 @@ class EmailAPI with HandleSetErrorMixin {
     List<Email> emails,
     ReadActions readActions
   ) async {
-    final setEmailMethod = SetEmailMethod(accountId)
-      ..addUpdates(emails.listEmailIds.generateMapUpdateObjectMarkAsRead(readActions));
+    final maxBatches = _getMaxObjectsInSetMethod(session, accountId);
+    final totalEmails = emails.length;
 
-    final getEmailMethod = GetEmailMethod(accountId)
-      ..addIds(emails.listEmailIds.toIds().toSet())
-      ..addProperties(Properties({EmailProperty.keywords}));
+    final List<Email> updatedEmails = List.empty(growable: true);
 
-    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    for (int start = 0; start < totalEmails; start += maxBatches) {
+      int end = (start + maxBatches < totalEmails)
+        ? start + maxBatches
+        : totalEmails;
+      log('EmailAPI::markAsRead:emails from ${start + 1} to $end');
 
-    requestBuilder.invocation(setEmailMethod);
+      final currentListEmails = emails.sublist(start, end);
 
-    final getEmailInvocation = requestBuilder.invocation(getEmailMethod);
+      final setEmailMethod = SetEmailMethod(accountId)
+        ..addUpdates(currentListEmails.listEmailIds.generateMapUpdateObjectMarkAsRead(readActions));
 
-    final capabilities = setEmailMethod.requiredCapabilities
-      .toCapabilitiesSupportTeamMailboxes(session, accountId);
+      final getEmailMethod = GetEmailMethod(accountId)
+        ..addIds(emails.listEmailIds.toIds().toSet())
+        ..addProperties(Properties({EmailProperty.keywords}));
 
-    final response = await (requestBuilder
-        ..usings(capabilities))
-      .build()
-      .execute();
+      final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
 
-    final getEmailResponse = response.parse<GetEmailResponse>(
-      getEmailInvocation.methodCallId,
-      GetEmailResponse.deserialize);
+      requestBuilder.invocation(setEmailMethod);
+
+      final getEmailInvocation = requestBuilder.invocation(getEmailMethod);
+
+      final capabilities = setEmailMethod.requiredCapabilities
+          .toCapabilitiesSupportTeamMailboxes(session, accountId);
+
+      final response = await (requestBuilder
+          ..usings(capabilities))
+        .build()
+        .execute();
```

Comment:
how about still logging setError?

Reply:
handled SetError and returned it in the function